config/300-gateway.yaml

```diff
@@ -85,7 +90,27 @@ spec:
           lifecycle:
             preStop:
               exec:
-                command: ["/bin/sh","-c","curl -X POST --unix /tmp/envoy.admin http://localhost/healthcheck/fail; sleep 15"]
+                command:
+                - perl
```

Comment:
I like that this is more compact but I feel that we should stick to bash so it is aligned with the rest of the codebase.

Reply:
Yep I understand, I have no strong feeling here, so I'll stick with what you prefer :grin:

To be fair, I've tried the `perl` approach, because some people are allergic to "big" shell scripts :grin: if you prefer the shell approach, I've already made this: https://github.com/knative-extensions/net-kourier/compare/main...norbjd:5be667d697ba34c65946ed42bf4b9533cc27b4c5

The only downside is we need to "open" the `/healtcheck/fail` or `/drain_listeners` endpoint on Envoy's admin interface, because there is no way I know of connecting to the `/tmp/envoy.admin` socket with bash utilities embedded in Envoy's image.

Reply:
Hello, I've reverted to pure shell: https://github.com/knative-extensions/net-kourier/pull/1203/commits/423a6da5f2255f73ff844f06ae5692ff4dc64bdc

Tell me what you think :)